packages/plugins/src/hardhat/deployments.ts

```diff
@@ -336,23 +335,32 @@ export const proposeChugSplashBundle = async (
   configUri: string,
   remoteExecution: boolean,
   ipfsUrl: string,
-  isUpgrade: boolean,
-  configPath: string
+  configPath: string,
+  spinner: ora.Ora
 ) => {
+  // Determine if the deployment is an upgrade
+  const isUpgrade = await checkIsUpgrade(hre.ethers.provider, parsedConfig)
   if (isUpgrade) {
+    spinner.succeed('Upgrade detected')
+    // Check if upgrade is valid
     await checkValidUpgrade(
       hre.ethers.provider,
       parsedConfig,
       configPath,
       hre.network.name
     )
-  } else {
-    await checkValidDeployment(
-      hre.ethers.provider,
-      parsedConfig,
-      configPath,
-      hre.network.name
-    )
+
+    // Confirm upgrade with user
+    const confirm = await yesno({
+      question: `Prior deployments detected for project ${parsedConfig.options.projectName}, would you like to perform an upgrade? (y/n)`,
+    })
```

Comment:
I like the idea of confirming with the user before upgrading, but this confirmation could be troublesome if performing an upgrade in a CI process. Perhaps we add a `--confirm` flag that allows the user to skip this?

Reply:
Yeah, I can add that. Though I do think we'll ultimately want a separate task specifically intended for use in CI processes.